cmd/daemonset-check/input.go

```diff
@@ -17,6 +17,9 @@ func parseInputValues() {
 	if err != nil {
 		log.Infoln("There was an issue getting the check deadline:", err.Error())
 	}
+	checkTimeout = timeDeadline.Sub(time.Now().Add(time.Second * 5))
```

Comment:
I think we should better identify what our time barriers are here...

- `checkTimeLimit` is how long we want this pod to run before considering the check a failure
- `timeDeadline` is the point when the checker pod is "dead" in the eyes of KH. no status can be reported after this time.
- `checkTimeout` is when the check officially considers the check as a failure and begins reporting it

This seems a little too complex.  We also have a `defaultCheckTimeLimit` variable and a `defaultShutdownGracePeriod`.  



How about we simplify the times here into this:

- the time at which the check will be orphaned by Kuberhealthy (`timeDeadline`)
- the time at which the check needs needs to consider its self failed and report a failure 

The orphaned time is simply `kh.GetDeadline()`.  The time at which the check needs to fail is that time, minus the graceful shutdown period.  Something like:

```
shutdownGracePeriod := time.Minute
podDeadline, err := kh.GetDeadline()
if err != nil {
    // stuff
}
checkDeadline := podDeadline.Add(-shutdownGracePeriod)
```

perhaps we could simplify to this?

Reply:
After you have `checkDeadline`, you can do stuff like this:

```go
if time.Now().After(checkDeadline) {
    // failure
}
```

Reply:
sounds good!